control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -141,6 +152,8 @@ type reconciler struct {
 	versions                  map[string]string
 	operateOnReleaseImage     string
 	ImageMetaDataProvider     util.ImageMetadataProvider
+	cleanupFailures           map[string]*cleanupFailureTracker
```

Comment:
the failure tracker (and associated mutex) should be in its own struct (likely under support/util).